src/coreclr/runtime/amd64/WriteBarriers.S

```diff
@@ -2,7 +2,15 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 .intel_syntax noprefix
-#include <unixasmmacros.inc>
+#include "AsmMacros_Shared.h"
+
+#if defined(__APPLE__)
+    // Currently the build is failing without this due to an issue if the first method in the assembly file has an alternate entry at the start of the file.
```

Comment:
Why is this not failing in NativeAOT build today?

Reply:
Now that I look at it, this probably isn't necessary in the amd64 write barrier since it wasn't complaining before.

Reply:
This leads me to wonder if maybe the issue on MacOS arm64 with having an alternate entry as the first symbol in the object file was that since our alternate entry macro doesn't set the thumb bit, the offset for the alternate entry was before that of the normal function symbol for the first symbol in the object file. Time for an experiment. Sigh.

Reply:
Thumb bit is only relevant for arm32 so that should not matter here.